packages/shared/src/utils.ts

```diff
@@ -184,13 +184,11 @@ export function getDims(dataArray: NdArray): Dims {
   return { rows, cols };
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function createMemo<P extends any[], R, T extends (...args: P) => R>(
-  fn: T
-): T;
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function createMemo<P extends any[], R, T extends (...args: P) => R>(
-  fn: T
-) {
-  return (...args: P) => useMemo<R>(() => fn(...args), args); // eslint-disable-line react-hooks/exhaustive-deps
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export function createMemo<T extends (...args: any[]) => any>(fn: T): T;
+
+export function createMemo<T extends (...args: any[]) => any>(fn: T) {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,react-hooks/exhaustive-deps
+  return (...args: any[]) => useMemo<any>(() => fn(...args), args);
 }
+/* eslint-enable @typescript-eslint/no-explicit-any */
```

Comment:
Might be worth putting `createMemo` in its own file to allow to use the disable of rules by file